### ISSUE TITLE
Connection reset by peer

### ISSUE THREAD

**Issue by martin5696** [soachishti/moss.py#18]
I keep getting this error when I try to run the script. The input size is large, could this be the problem? I don't think it's a connectivity issue because it works on smaller input size.
Thanks!

Traceback (most recent call last):
  File "moss_script.py", line 370, in <module>
    main()
  File "moss_script.py", line 203, in main
    run_mosspy(args, args.language, base_files, student_files)
  File "moss_script.py", line 269, in run_mosspy
    url = moss.send()
  File "/homes/z/zhan1486/.local/lib/python2.7/site-packages/mosspy/moss.py", line 129, in send
    self.uploadFile(s, file_path, display_name, index)
  File "/homes/z/zhan1486/.local/lib/python2.7/site-packages/mosspy/moss.py", line 103, in uploadFile
    s.send(message.encode())
socket.error: [Errno 104] Connection reset by peer

**Comment by llayman** [comment]
This MOSS server at Stanford was having some issues last week. Do you still have the problem?

**Comment by BingqingWei** [comment]
I'm still having this problem...

**Comment by naquiroz** [comment]
Same here.

**Comment by soachishti** [comment]
There was a similar issue #9, can you verify that your network can communicate with moss.stanford.edu at port on 7690?

**Comment by naquiroz** [comment]
Verified. I'm sorry, the error was that I forgot to change de ID . Thanks!

**Comment by soachishti** [comment]
Np, thanks for the update. @naquiroz 

@BingqingWei ?

**Comment by andrewjbennett** [comment]
I'm also having this problem. My connection gets reset after almost exactly 10 minutes.

The same thing happens using the official moss upload script, except that moss.py is able to get through more of the submissions (~20% of them, vs ~10-15% with mossnet).

I added some logging with timestamps, here's a graph of the uploads it manages to do over time:

![image](https://user-images.githubusercontent.com/864138/57122765-56de4100-6dc2-11e9-9b38-644c372e04cb.png)

The very first timestamp is when it sends the first request to moss (which takes about a minute from sending it to when moss responds saying ok); subsequent timestamps are when each file upload finished.

I would guess that it's just a load problem with moss (except that it seems like the load never goes away), but perhaps there was some sort of configuration/server change at some point in the past -- the last time I was able to successfully upload to moss was approx October 2018 (not that I've been trying continually since then, but wasn't able to complete an upload in late 2018, nor early 2019, nor now.).

I too am sending a rather large upload: ~700 .c files, with a total size of ~15MB.
